src/analysis/special_functions/gamma.lean

```diff
@@ -127,3 +133,204 @@ begin
+    rw partial_Gamma_recurrence hs (mem_Ici.mp hX),
+    ring_nf, },
+  refine tendsto.congr' this _,
+  suffices : tendsto (λ X, -X ^ s * (-X).exp : ℝ → ℂ) at_top (𝓝 0),
+  { simpa using tendsto.add (tendsto.const_mul s (tendsto_partial_Gamma hs)) this },
+  rw tendsto_zero_iff_norm_tendsto_zero,
+  have : (λ (e : ℝ), ∥-(e:ℂ) ^ s * (-e).exp∥ ) =ᶠ[at_top] (λ (e : ℝ), e ^ s.re * (-1 * e).exp ),
+  { refine eventually_eq_of_mem (Ioi_mem_at_top 0) _,
+    intros x hx, dsimp only,
+    rw [norm_eq_abs, abs_mul, abs_neg, abs_cpow_eq_rpow_re_of_pos hx,
+      abs_of_nonneg (exp_pos(-x)).le, neg_mul, one_mul],},
+  exact (tendsto_congr' this).mpr (tendsto_rpow_mul_exp_neg_mul_at_top_nhds_0 _ _ zero_lt_one),
+end
+
+end Gamma_recurrence
+
+/- Now we define `Γ(s)` on the whole complex plane, by recursion. -/
+
+section Gamma_def
+
+/-- This function is `Γ(s)` if `1-n ≤ s.re`, and junk otherwise. -/
+noncomputable def Gamma_aux : ℕ → (ℂ → ℂ)
+| 0      := Gamma_integral
+| (n+1)  := λ s:ℂ, (Gamma_aux n (s+1)) / s
+
+lemma Gamma_aux_recurrence1 (s : ℂ) (n : ℕ) (h1 : 1 - s.re ≤ ↑n) :
+  Gamma_aux n s = Gamma_aux n (s+1) / s :=
+begin
+  induction n with n hn generalizing s,
+  { simp only [nat.cast_zero, sub_nonpos] at h1,
+    dsimp only [Gamma_aux], rw Gamma_integral_recurrence h1,
+    rw [mul_comm, mul_div_cancel], contrapose! h1, rw h1,
+    simp },
+  { dsimp only [Gamma_aux],
+    have hh1 : 1 - (s+1).re ≤ n,
+    { rw [nat.succ_eq_add_one, nat.cast_add, nat.cast_one] at h1,
+      rw [add_re, one_re], linarith, },
+    rw ←(hn (s+1) hh1) }
+end
+
+lemma Gamma_aux_recurrence2 (s : ℂ) (n : ℕ) (h1 : 1 - s.re ≤ ↑n) :
+  Gamma_aux n s = Gamma_aux (n+1) s :=
+begin
+  cases n,
+  { simp only [nat.cast_zero, sub_nonpos] at h1,
+    dsimp only [Gamma_aux], rw Gamma_integral_recurrence h1,
+    have : s ≠ 0 := by { contrapose! h1, rw h1, simp, },
+    field_simp, ring },
+  { dsimp only [Gamma_aux],
+    have : (Gamma_aux n (s + 1 + 1)) / (s+1) = Gamma_aux n (s + 1),
+    { have hh1 : 1 - (s+1).re ≤ n,
+      { rw [nat.succ_eq_add_one, nat.cast_add, nat.cast_one] at h1,
+        rw [add_re, one_re], linarith, },
+      rw Gamma_aux_recurrence1 (s+1) n hh1, },
+    rw this },
+end
+
+/-- The `Γ` function (of a complex variable `s`). -/
+def Gamma (s : ℂ) : ℂ := Gamma_aux ⌈ 1 - s.re ⌉₊ s
+
+lemma Gamma_eq_Gamma_aux (s : ℂ) (n : ℕ) (h1 : 1 - s.re ≤ ↑n) : Gamma s = Gamma_aux n s :=
+begin
+  have u : ∀ (k : ℕ), Gamma_aux (⌈ 1 - s.re ⌉₊ + k) s = Gamma s,
+  { intro k, induction k with k hk,
+    { simp [Gamma],},
+    { rw [←hk, nat.succ_eq_add_one, ←add_assoc],
+      refine (Gamma_aux_recurrence2 s (⌈ 1 - s.re ⌉₊ + k) _).symm,
+      rw nat.cast_add,
+      have i1 := nat.le_ceil (1 - s.re),
+      refine le_add_of_le_of_nonneg i1 _,
+      rw [←nat.cast_zero, nat.cast_le], exact nat.zero_le k, } },
+  rw [←nat.add_sub_of_le (nat.ceil_le.mpr h1), u (n - ⌈ 1 - s.re ⌉₊)],
+end
+
+theorem Gamma_recurrence (s : ℂ) (h2 : s ≠ 0) : s * Gamma(s) = Gamma (s+1) :=
```

Comment:
What do you think will be the most common way that people will rewrite with this theorem? My guess would be from-right-to-left, and so intuitively I would state it in the other direction. But maybe my intuition is wrong!

Reply:
You're totally right, as it's written it is inconsistent with elsewhere in the file. I will fix it in a moment.